server/src/main/java/io/crate/execution/engine/sort/LuceneSort.java

```diff
@@ -152,10 +153,26 @@ public SortField visitReference(Reference ref, final SortSymbolContext context)
             return customSortField(ref.toString(), ref, context);
         }
 
+        if (ref.valueType() instanceof NumericType numericType) {
```

Comment:
Maybe I'm missing something, why can't we use the `mappedSortField` in this case?

Reply:
For the `precision == null || > 18` case the numeric sort field can't be used, so there's a `customSortField` return. I can't do that inside `mappedSortField` because that's static and `customSortfield` is not. 

And it would need a dedicated switch, because the nullSentinel and the `setOptimizeSortWithPoints` aspects are different.

Could probably be reshuffled somewhat, and I might try following up after https://github.com/crate/crate/pull/16505 to pull out the whole `mappedSortField` switch and merge it with the IP/BitString/FloatVector into a single pattern matching switch but I don't think either variant makes a big difference in readability or maintainability.

Reply:
thank you for explaining!